libs/utils/src/http/endpoint.rs

```diff
@@ -32,6 +32,36 @@ static X_REQUEST_ID_HEADER: HeaderName = HeaderName::from_static(X_REQUEST_ID_HE
 #[derive(Debug, Default, Clone)]
 struct RequestId(String);
 
+/// Adds a tracing info_span! instrumentation around the handler events.
+/// Use this to distinguish between logs of different HTTP requests.
+pub struct RequestSpan<B, E, R, H>(pub H)
+where
+    B: hyper::body::HttpBody + Send + Sync + 'static,
+    E: Into<Box<dyn std::error::Error + Send + Sync>> + 'static,
+    R: Future<Output = Result<Response<B>, E>> + Send + 'static,
+    H: Fn(Request<Body>) -> R + Send + Sync + 'static;
+
+impl<B, E, R, H> RequestSpan<B, E, R, H>
+where
+    B: hyper::body::HttpBody + Send + Sync + 'static,
+    E: Into<Box<dyn std::error::Error + Send + Sync>> + 'static,
+    R: Future<Output = Result<Response<B>, E>> + Send + 'static,
+    H: Fn(Request<Body>) -> R + Send + Sync + 'static,
+{
+    /// Creates a tracing span around inner request handler and executes the request handler in the contex of that span.
+    /// Use as `|r| RequestSpan(my_handler).handle(r)` instead of `my_handler` as the request handler to get the span enabled.
+    pub async fn handle(self, request: Request<Body>) -> Result<Response<B>, E> {
+        let request_id = request.context::<RequestId>().unwrap_or_default().0;
+
+        let method = request.method().to_owned();
+        let path = request.uri().path().to_owned();
+
+        (self.0)(request)
+            .instrument(info_span!("request", %method, %path, %request_id))
+            .await
```

Comment:
So if the span would be created before we move the request to the inner handler, these would not have to be owned, and span would handle the formatting from borrowed value just as well as from owned value.

Is there some downside in eagerly creating the span?

```suggestion
        let request_id = request.context::<RequestId>().unwrap_or_default().0;

        let method = request.method();
        let path = request.uri().path();
        
        let span = info_span!("request", %method, %path, %request_id);

        (self.0)(request)
            .instrument(span)
            .await
```

Reply:
Thanks for noticing, fixed.